### PR TITLE
Update dependencies to Sonata 4

### PR DESCRIPTION
**Pull request by wbloszyk**
## Subject

I am targeting this branch, because this is BC break

## Changelog

```markdown
### Added
- Add Compatibility with Sonata 4

### Changed
- `Sonata\NewsBundle\Admin\PostAdmin::__constructor` will now require `sonata.admin.configuration` service as 4 argument
```

## To do

- [ ] https://github.com/sonata-project/dev-kit/pull/1698

### CODE REVIEW

File: src/Admin/PostAdmin.php [hunk 1]
```diff
@@ -205,23 +200,29 @@ protected function configureTabMenu(MenuItemInterface $menu, $action, ?AdminInte
...
-                ['uri' => $admin->getRouteGenerator()->generate(
-                    'sonata_news_view',
-                    ['permalink' => $this->permalinkGenerator->generate($this->getSubject())]
-                )]
-            );
+        if (!$this->hasSubject()) {
+            return;
+        }
+
+        if ($subject = $this->getSubject()) {
```
Comment (VincentLanglet) [inline src/Admin/PostAdmin.php]: Useless if

File: src/Block/RecentCommentsBlockService.php [hunk 2]
```diff
@@ -14,60 +14,47 @@
...
+final class RecentCommentsBlockService extends AbstractBlockService implements EditableBlockService
 {
     /**
      * @var CommentManagerInterface
      */
     protected $manager;
 
     /**
-     * @var Pool
+     * @var ?Pool
```
Comment (VincentLanglet) [inline src/Block/RecentCommentsBlockService.php]: Pool|null, but it could be made required

File: src/Block/RecentPostsBlockService.php [hunk 3]
```diff
@@ -14,60 +14,47 @@
...
+final class RecentPostsBlockService extends AbstractBlockService implements EditableBlockService
 {
     /**
      * @var PostManagerInterface
      */
     protected $manager;
 
     /**
-     * @var Pool
+     * @var ?Pool
```
Comment (VincentLanglet) [inline src/Block/RecentPostsBlockService.php]: Same

File: composer.json [hunk 4]
```diff
@@ -22,14 +22,14 @@
...
+        "php": "^7.4 || ^8.0",
         "doctrine/persistence": "^1.3 || ^2.0",
-        "sonata-project/classification-bundle": "^3.12",
+        "sonata-project/classification-bundle": "^4.0@dev",
         "sonata-project/doctrine-extensions": "^1.10.1",
         "sonata-project/form-extensions": "^1.4",
         "sonata-project/formatter-bundle": "^4.0",
         "sonata-project/intl-bundle": "^2.4",
-        "sonata-project/media-bundle": "^3.20",
+        "sonata-project/media-bundle": "^4.0@dev",
```
Comment (VincentLanglet) [inline composer.json:32]: We have to first release the major version of the media bundle and the classification bundle then

Comment (wbloszyk) [reply composer.json:32]: Agree. Anyway we can add `^4.0@dev` or `4.x-dev` in unstable branch here to allow work on Sonata 4 in all sonata dependencies at once. We have to remove it before stable release.

Comment (VincentLanglet) [reply composer.json:32]: Sure

**Review by ChristianGripp — DISMISSED** [review]

File: src/Admin/PostAdmin.php [hunk 5]
```diff
@@ -48,26 +45,29 @@ class PostAdmin extends AbstractAdmin
...
     protected $permalinkGenerator;
-
     /**
-     * @deprecated since sonata-project/news-bundle 3.13, to be removed in 4.0.
+     * @var SonataConfiguration
      */
-    public function setUserManager($userManager): void
+    private $sonataConfiguration;
+
+    public function __construct(string $code, string $class, string $baseControllerName, SonataConfiguration $sonataConfiguration)
```
Comment (VincentLanglet) [inline src/Admin/PostAdmin.php:53]: I wonder if you have something better in mind than injecting the whole configuration @franmomu

Comment (franmomu) [reply src/Admin/PostAdmin.php:53]: nope 😞 I think this option is not in the container as parameter.

**Pull request closed without merging**